packages/relic/lib/src/headers/headers.dart

```diff
@@ -0,0 +1,1688 @@
+        _acceptEncodingHeader,
+        onParse: AcceptEncodingHeader.parse,
+      ),
+      acceptLanguage: dartIOHeaders.parseMultipleValue(
+        _acceptLanguageHeader,
+        onParse: AcceptLanguageHeader.parse,
+      ),
+      accessControlRequestHeaders: dartIOHeaders.parseStringList(
+        _accessControlRequestHeadersHeader,
+      ),
+      accessControlRequestMethod: dartIOHeaders.parseSingleValue(
+        _accessControlRequestMethodHeader,
+        onParse: Method.parse,
+      ),
+      age: dartIOHeaders.parseInt(
+        _ageHeader,
+        allowNegative: false,
+      ),
+      allow: dartIOHeaders.parseMultipleValue(
+        _allowHeader,
+        onParse: (values) {
+          return values
+              .splitTrimAndFilterUnique(emptyCheck: false)
+              .map(Method.parse)
+              .toList();
+        },
+      ),
+      cookie: dartIOHeaders.parseSingleValue(
+        _cookieHeader,
+        onParse: CookieHeader.parse,
+      ),
+      setCookie: dartIOHeaders.parseSingleValue(
+        _setCookieHeader,
+        onParse: CookieHeader.parse,
+      ),
+      authorization: dartIOHeaders.parseSingleValue(
+        _authorizationHeader,
+        onParse: AuthorizationHeader.parse,
+      ),
+      connection: dartIOHeaders.parseMultipleValue(
+        _connectionHeader,
+        onParse: ConnectionHeader.parse,
+      ),
+      expect: dartIOHeaders.parseSingleValue(
+        _expectHeader,
+        onParse: ExpectHeader.parse,
+      ),
+      ifMatch: dartIOHeaders.parseMultipleValue(
+        _ifMatchHeader,
+        onParse: IfMatchHeader.parse,
+      ),
+      ifNoneMatch: dartIOHeaders.parseMultipleValue(
+        _ifNoneMatchHeader,
+        onParse: IfNoneMatchHeader.parse,
+      ),
+      ifRange: dartIOHeaders.parseSingleValue(
+        _ifRangeHeader,
+        onParse: IfRangeHeader.parse,
+      ),
+      maxForwards: dartIOHeaders.parseInt(
+        _maxForwardsHeader,
+        allowNegative: false,
+      ),
+      proxyAuthorization: dartIOHeaders.parseSingleValue(
+        _proxyAuthorizationHeader,
+        onParse: AuthorizationHeader.parse,
+      ),
+      range: dartIOHeaders.parseSingleValue(
+        _rangeHeader,
+        onParse: RangeHeader.parse,
+      ),
+      referer: dartIOHeaders.parseUri(_refererHeader),
+      te: dartIOHeaders.parseMultipleValue(
+        _teHeader,
+        onParse: TEHeader.parse,
+      ),
+      upgrade: dartIOHeaders.parseMultipleValue(
+        _upgradeHeader,
+        onParse: UpgradeHeader.parse,
+      ),
+      userAgent: dartIOHeaders.parseString(
+        _userAgentHeader,
+      ),
+
+      // Response Headers
+      location: dartIOHeaders.parseUri(_locationHeader),
+      xPoweredBy: dartIOHeaders.parseString(
+            _xPoweredByHeader,
+          ) ??
+          xPoweredBy,
+      accessControlAllowCredentials: dartIOHeaders.parseBool(
+        _accessControlAllowCredentialsHeader,
+        allowFalse: false,
+      ),
```

Comment:
Suggestion: Create a specialized method to parse this.

Strange to only allow positive values. Should negative values be discarded rather than throwing?

Reply:
Yes only positive, but you are right, we can set it to null instead of throwing 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials#directives